contracts/lib/LibLiquidation.sol

```diff
@@ -138,4 +161,32 @@ library LibLiquidation {
             return amountToReturn;
         }
     }
+
+    /**
+     * @return true if the margin is greater than 10x liquidation gas cost (in quote tokens)
+     * @dev Assumes params are WAD except liquidationGasCost
+     * @param liquidatedBaseChange How much base token to be liquidated
+     * @param liquidatedQuoteChange How much quote token to be liquidated
+     * @param balanceToBeLiquidated The balance of account to be liquidated
+     * @param liquidationGasCost Approximately how much gas is used to call liquidate()
+     * @param price Current fair price
+     */
+    function partialLiquidationIsValid(
+        int256 liquidatedBaseChange,
+        int256 liquidatedQuoteChange,
+        Balances.Account memory balanceToBeLiquidated,
+        uint256 liquidationGasCost,
+        uint256 price
+    ) internal pure returns (bool) {
+        uint256 minimumLeftoverMargin = PRBMathUD60x18.mul(
+            balanceToBeLiquidated.lastUpdatedGasPrice,
+            liquidationGasCost
+        ) * 10;
+        Balances.Position memory updatedPosition = Balances.Position(
+            balanceToBeLiquidated.position.quote + liquidatedQuoteChange,
+            balanceToBeLiquidated.position.base + liquidatedBaseChange
+        );
+        int256 margin = Balances.margin(updatedPosition, price);
+        return margin >= minimumLeftoverMargin.toInt256() || margin == 0;
```

Comment:
Bit worried about this `margin == 0` check. Technically margin as 0 could be a valid partial liquidation point. Should we also check the user has positions or something rather than just `margin == 0`. Also worth seeing if you can even get to this check with zero positions (base)

Reply:
Yeah I think this should actually be return `margin >= minimumLeftoverMargin.toInt256() || updatedPosition.base == 0;`

Is that what you're saying?

Reply:
Actually, it will need to be `margin >= minimumLeftoverMargin.toInt256() || (updatedPosition.base == 0 && updatedPosition.quote == 0);` since if base == 0, quote can be negative and this position is still liquidatable.